### ISSUE TITLE
SortedSet.collect does not accept a function for which there is no Ordering on the codomain

### ISSUE THREAD

**Issue by MasseGuillaume** [scala/bug#11019]
2.12

```scala
case class A(v: Int)
collection.immutable.SortedSet(1).collect{ case x => A(x) }
// collection.immutable.Set[A] = Set(A(1))
```

2.13

```scala
[error] /home/gui/scala/test/junit/scala/Playground.scala:12: No implicit Ordering defined for A.
[error]     collection.immutable.SortedSet(1).collect{ case x => A(x) }
[error]                                              ^
```

**Comment by MasseGuillaume** [comment]
Hum, this is probably due to the absence of `CanBuildFrom`.

For `map`, I think the behavior it 2.13 make sense:

2.12

```scala
immutable.SortedSet(1).map(x => x)
// collection.immutable.SortedSet[Int] = TreeSet(1)

immutable.SortedSet(1).map(x => A(x))
// collection.immutable.Set[A] = Set(A(1))
```

2.13
```
[error] /home/gui/scala/test/junit/scala/Playground.scala:12: No implicit Ordering defined for A.
[error]     collection.immutable.SortedSet(1).map(x => A(x))
[error]                                          ^

**Comment by MasseGuillaume** [comment]
This is documented here: https://github.com/scala/collection-strawman/wiki/FAQ#what-are-the-breaking-changes.

workarround:
val temp: Set[Int] = collection.immutable.SortedSet(1).
temp.collect{ case x => A(x) }

**Comment by julienrf** [comment]
What should we do @MasseGuillaume?

This is a documented change and the recommended way to achieve what you want to achieve in the first post is to use the `unsorted` operation:

~~~ scala
case class A(v: Int)
collection.immutable.SortedSet(1).unsorted.collect{ case x => A(x) }
// collection.immutable.Set[A] = Set(A(1))
~~~

**Comment by MasseGuillaume** [comment]
`unsorted` is only available in 2.13: https://github.com/scala/scala-collection-compat/issues/95
